File: ocaml-server/tests/test_routing.py
"""
Test basic HTTP routing: methods, unknown paths, content types.
"""

import requests


class TestRouting:
    """Verify that the server routes requests correctly and rejects bad ones."""

    def test_get_root_returns_405(self, base_url):
        """GET on any path should return 405 Method Not Allowed (server is POST-only)."""
        r = requests.get(base_url + "/", timeout=10)
        assert r.status_code == 405

    def test_get_ingest_returns_405(self, base_url):
        r = requests.get(base_url + "/ingest", timeout=10)
        assert r.status_code == 405

    def test_post_unknown_path_returns_404(self, base_url):
        r = requests.post(base_url + "/nonexistent", timeout=10)
        assert r.status_code == 404

    def test_post_unknown_nested_path_returns_404(self, base_url):
        r = requests.post(base_url + "/admin/nonexistent", timeout=10)
        assert r.status_code == 404

File: rag-o-mail/test/test_routing.ml
open Helpers

let test_get_root_returns_405 () =
  skip_if_unreachable ();
  let code, _ = get ~path:"/" in
  Alcotest.(check int) "GET / → 405" 405 code

let test_get_ingest_returns_405 () =
  skip_if_unreachable ();
  let code, _ = get ~path:"/ingest" in
  Alcotest.(check int) "GET /ingest → 405" 405 code

let test_post_unknown_path_returns_404 () =
  skip_if_unreachable ();
  let code, _ = post_json ~path:"/nonexistent" ~body_str:"{}" in
  Alcotest.(check int) "POST /nonexistent → 404" 404 code

let test_post_unknown_nested_path_returns_404 () =
  skip_if_unreachable ();
  let code, _ = post_json ~path:"/admin/nonexistent" ~body_str:"{}" in
  Alcotest.(check int) "POST /admin/nonexistent → 404" 404 code

let tests =
  [ Alcotest.test_case "GET / → 405"                    `Quick test_get_root_returns_405
  ; Alcotest.test_case "GET /ingest → 405"              `Quick test_get_ingest_returns_405
  ; Alcotest.test_case "POST unknown → 404"             `Quick test_post_unknown_path_returns_404
  ; Alcotest.test_case "POST admin/unknown → 404"       `Quick test_post_unknown_nested_path_returns_404
  ]
